spice/lib/vm_actions_rv.py

```diff
@@ -350,23 +350,24 @@ def rv_chk_con(vmi):
         hostname = socket.gethostname()    # See rv_url() function
         remote_ip = socket.gethostbyname(hostname)
     elif cfg.spice_proxy:
-        proxy_port = "3128"
-        if "http" in cfg.spice_proxy:
-            split = cfg.spice_proxy.split('//')[1].split(':')
+        prx_parse = cfg.spice_proxy.split('//')[-1]
+        if ']:' in prx_parse or '.' in prx_parse and ':' in prx_parse:
```

Comment:
Maybe put here example in comment to easy understand what kind of info your are trying to parse:

# If prx_parse has form: http://[ipv6]:3218